File: Java/CountSort.java
import java.util.ArrayList;
import java.util.InputMismatchException;
import java.util.Scanner;

public class CountSort {
    public static void main(String[] args) {
        ArrayList<Integer> arr = new ArrayList<>();
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter size of the list: ");
        try {
            int size = scanner.nextInt();

            if (size <= 0) {
                System.out.println("Size of list can not be negative or zero");
                scanner.close();
                return;
            }
            System.out.println("Enter the elements: ");
            for (int i = 0; i < size; i++) {
                int el = scanner.nextInt();
                arr.add(el);
            }
            System.out.println("INPUT LIST: ");
            for (int num : arr)
                System.out.print(num + "\t");
            countSort(arr);
            System.out.println("\nSORTED LIST: ");
            for (int num : arr)
                System.out.print(num + "\t");
        } catch (InputMismatchException e) {
            System.out.println("INPUT MUST BE IN NUMBERS");
        }
        scanner.close();
    }

    public static void countSort(ArrayList<Integer> arr) {
        if (arr == null || arr.isEmpty()) {
            return;
        }
        int max = Integer.MIN_VALUE;
        int min = Integer.MAX_VALUE;

        for (Integer num : arr) {
            if (num > max) {
                max = num;
            }
            if (num < min) {
                min = num;
            }
        }
        int[] countArray = new int[max - min + 1];
        for (Integer num : arr) {
            countArray[num - min]++;
        }

        int index = 0;
        for (int i = 0; i < countArray.length; i++) {
            while (countArray[i] > 0) {
                arr.set(index++, i + min);
                countArray[i]--;
            }
        }
    }

}

File: Python/CountSort.py
def CountSort(arr):
    high = max(arr)
    freq = [0] * (high + 1)
    for num in arr:
        freq[num] += 1
    for i in range(1, high + 1):
        freq[i] += freq[i - 1]
    
    sorted_list = [0] * len(arr)
    
    for i in range(len(arr) - 1, -1, -1):
        sorted_list[freq[arr[i]] - 1] = arr[i]
        freq[arr[i]] -= 1
    
    return sorted_list

try:
    arr = []
    size = int(input("Enter the size of the list: "))
    for _ in range(size):
        while True:
            try:
                el = int(input("Enter a number: "))
                arr.append(el)
                break 
            except ValueError:
                print("Invalid input. Please enter a valid number.")

    print("INPUT LIST:", arr)
    arr = CountSort(arr)
    print("SORTED LIST:", arr)
except KeyboardInterrupt:
    print("Program interrupted by the user.")
except Exception as e:
    print(f"INPUT MUST BE IN NUMBER")